changelog/2.076.0_pre.dd

```diff
@@ -0,0 +1,374 @@
+)
+
+---
+import std.algorithm.comparison : equal;
+
+int i;
+
+// The generator doesn't save state, so it cannot be a forward range.
+auto inputRange = generate!(() => ++i).take(10);
+
+// We can still process it in chunks, but it will be single-pass only.
+auto chunked = inputRange.chunks(2);
+
+assert(chunked.front.equal([1, 2]));
+assert(chunked.front.empty); // Iterating the chunk has consumed it
+chunked.popFront;
+assert(chunked.front.equal([3, 4]));
+---
+)
+
+$(LI $(LNAME2 std-socket-abstract,`std.socket.UnixAddress` now supports abstract addresses.)
+$(P
+UNIX domain sockets are usually identified by pathnames. Linux offers a
+non-portable extension to this scheme, known as abstract socket addresses,
+which are independent of the filesystem. An abstract socket address starts with
+a null byte (`'\0'`), e.g.:
+)
+
+---
+auto addr = new UnixAddress("\0/tmp/dbus-OtHLWmCLPR");
+---
+)
+
+$(LI $(LNAME2 std-typecons-AutoImplement,Added possibility to use a baseclass when auto-implementing an interface)
+$(P
+A second $(REF AutoImplement, std, typecons) template has been added, which
+differs from the existing one in accepting an extra type parameter. The extra
+parameter specifies a class to derive from while auto-implementing an interface.
+)
+
+$(P
+The base class used may contain non-default constructors. Matching constructors
+will be created in the auto-implemented class and be implemented as just a call
+to super with all arguments.
+)
+)
+
+
+)
+
+$(BUGSTITLE Tools changes,
+
+$(LI $(LNAME2 b11997,rdmd now checks for a D compiler binary in the directory it's in first)
+$(P
+If you downloaded a zip/tar file with the compiler before
+and ran `rdmd`, it would invoke the D compiler from the system
+`PATH` rather than the compiler right next to it, failing if
+there wasn't one in the `PATH`. `rdmd` will now try to use
+the D compiler next to it first, and only fall back to the
+`PATH` if there isn't one adjacent. If you want rdmd to run a
+specific compiler, add the `--compiler` flag to force it, as
+always.
+)
+
+$(P
+To restore the old behaviour of only using the compiler from
+your `PATH`, add `--compiler=dmd` (or `ldmd2` or `gdmd`).
+)
+)
+
+
+)
+
+$(BR)$(BIG $(LNAME2 bugfix-list, List of all bug fixes and enhancements in D $(VER):))
+
+$(BUGSTITLE DMD Compiler regressions,
+
+$(LI $(BUGZILLA 16680): dmd doesn't use druntime optimized versions of subtraction array operations)
+$(LI $(BUGZILLA 17522): win64.mak broken)
+$(LI $(BUGZILLA 17582): [REG2.059] Applying const to struct declaration should make the struct type const)
+$(LI $(BUGZILLA 17612): [REG2.063] Segmentation fault with bad object.d)
+$(LI $(BUGZILLA 17690): [REG2.066.0] scope guards leak declarations)
+$(LI $(BUGZILLA 17695): [Reg 2.076] ICE with vector negation)
+)
+$(BUGSTITLE DMD Compiler bugs,
+
+$(LI $(BUGZILLA 4014): CodeView debug type info not linked in from library)
+$(LI $(BUGZILLA 11881): -betterC switch suffers from bit rot)
```

Comment:
This bug is combination of many enhancement requests, only small portion of which were addressed for this release. I think we should reopen it. (The biggest unaddressed issue is that you still get link errors regarding TypeInfo, which exactly what the description of the bug talks about.)